tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -2016,3 +2019,36 @@ def test_functions_layers_with_s3_codeuri(self):
                 [""],  # there is only one stack
                 command_result,
             )
+
+
+@skipIf(SKIP_SAR_TESTS, "Skip SAR tests")
+class TestBuildSAR(BuildIntegBase):
+    template = "aws-serverless-application-with-application-id-map.yaml"
+
+    @parameterized.expand(
+        [
+            ("use_container", "us-east-2"),
+            ("use_container", "eu-west-1"),
+            ("use_container", None),
+            (False, "us-east-2"),
+            (False, "eu-west-1"),
+            (False, None),
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_sar_application_with_location_resolved_from_map(self, use_container, region):
```

Comment:
I won't block the PR on this, but I think it is really important to keep our tests in good shape.

The way I always think about testing is: if/when the test fails, can I extract enough information from the name and output to know what I broke? Generally this framing will lead to very focused tests that have a single outcome. It might mean we have almost identical tests but do/produce very different things. When I read this, I see two distinct use-cases: 1) When the command is suppose to pass, 2) when the command is suppose to fail and have a specific output in stderr. Since the test has additional logic to determine when to test 1 vs 2, it becomes very confusing to people changing code. They will then have to inspect and read the test to understand that `test_sar_application_with_location_resolved_from_map_0_False_1_eu-west-1` is suppose to fail when the region is not resolved. 

I generally believe that combining tests that have different outcomes is a complete anti-pattern and something we shouldn't be doing in general. I used to have some good links about this but am failing to find them now.

I don't mean to be this nit-picky on things but I strongly discourage this. I would prefer to have almost two identical test that validate 1 and 2 as very concrete things.